src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -93,12 +119,58 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
     }
 
     async deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._deinitializePollingTimer(this._pollingInterval);
         this._orderId = undefined;
         this._paypalCommercePaymentProcessor.deinitialize();
 
         return Promise.resolve(this._store.getState());
     }
 
+    private _initializePollingMechanism(submitForm: () => void, gatewayId?: string, timer?: number) {
+        if (!timer) {
```

Comment:
I believe it would be better to keep `timer` counter in the class property and initialize it with the default value (0). Along with the `_deinitializePollingTimer ` we can reset this value back to 0 if needed

Reply:
fixed